src/main/groovy/org/gradle/android/workarounds/room/KspWorkaround.groovy

```diff
@@ -0,0 +1,90 @@
+    }
+
+    @Override
+    void initWorkaround() {
+        project.tasks.withType(kspTaskClass).configureEach {
+            if (kspIsAppliedWithRoom()) {
+                configureWorkaroundTask(it)
+            }
+        }
+    }
+
+    @Override
+    void configureWorkaroundTask(Task task) {
+        def fileOperations = project.fileOperations
+        def variantSpecificSchemaDir = project.objects.directoryProperty()
+        variantSpecificSchemaDir.set(androidVariantProvider.getVariantSpecificSchemaDir(project, "${task.name}"))
+        task.commandLineArgumentProviders.add(new KspRoomSchemaLocationArgumentProvider(roomExtension.schemaLocationDir, variantSpecificSchemaDir))
+
+        task.doFirst onlyIfSymbolProcessorConfiguredForKsp(task.commandLineArgumentProviders) { KspRoomSchemaLocationArgumentProvider provider ->
+            KspWorkaround.copyExistingSchemasToTaskSpecificTmpDir(fileOperations, roomExtension.schemaLocationDir, provider)
+        }
+
+        task.doLast onlyIfSymbolProcessorConfiguredForKsp(task.commandLineArgumentProviders) { KspRoomSchemaLocationArgumentProvider provider ->
+            KspWorkaround.copyGeneratedSchemasToOutput(fileOperations, provider)
+        }
+
+        task.finalizedBy onlyIfSymbolProcessorConfiguredForKsp(task.commandLineArgumentProviders) {
+            roomExtension.schemaLocationDir.isPresent() ? mergeTask : null
+        }
+
+        TaskExecutionGraph taskGraph = project.gradle.taskGraph
+        taskGraph.whenReady onlyIfSymbolProcessorConfiguredForKsp(task.commandLineArgumentProviders) { KspRoomSchemaLocationArgumentProvider provider ->
+            if (taskGraph.hasTask(task)) {
+                roomExtension.registerOutputDirectory(provider.schemaLocationDir)
+            }
+        }
+    }
+
+    static void copyExistingSchemasToTaskSpecificTmpDir(FileOperations fileOperations, Provider<Directory> existingSchemaDir, KspRoomSchemaLocationArgumentProvider provider) {
+        if (existingSchemaDir.isPresent()) {
+            def temporaryVariantSpecificSchemaDir = provider.temporarySchemaLocationDir
+            fileOperations.sync {
+                it.from existingSchemaDir
+                it.into temporaryVariantSpecificSchemaDir
+            }
+        }
+    }
+
+    static void copyGeneratedSchemasToOutput(FileOperations fileOperations, KspRoomSchemaLocationArgumentProvider provider) {
+        def variantSpecificSchemaDir = provider.schemaLocationDir
+        def temporaryVariantSpecificSchemaDir = provider.temporarySchemaLocationDir
+        fileOperations.sync {
+            it.from temporaryVariantSpecificSchemaDir
+            it.into variantSpecificSchemaDir
+        }
+    }
+
+    private static Class<?> getKspTaskClass() {
+        return Class.forName("com.google.devtools.ksp.gradle.KspTaskJvm")
+    }
+
+    private static Closure onlyIfSymbolProcessorConfiguredForKsp(def commandLineArgumentProviders, Closure<?> action) {
```

Comment:
why is this named `onlyIfSymbolProcessorConfiguredForKsp` it seems like it is checking for the presence of a `KspRoomSchemaLocationArgumentProvider` ?

Reply:
done https://github.com/gradle/android-cache-fix-gradle-plugin/pull/429/commits/582f872d37fc1462346dea737da1e18479d85f50